File: app/src/main/java/de/markusfisch/android/shadereditor/hardware/PressureListener.java
package de.markusfisch.android.shadereditor.hardware;

import android.content.Context;
import android.hardware.Sensor;
import android.hardware.SensorEvent;

import androidx.annotation.NonNull;

public class PressureListener extends AbstractListener {
	private float pressure = 0f;

	public PressureListener(@NonNull Context context) {
		super(context);
	}

	public boolean register() {
		return register(Sensor.TYPE_PRESSURE);
	}

	@Override
	public void onSensorChanged(@NonNull SensorEvent event) {
		pressure = event.values[0];
	}

	public float getPressure() {
		return pressure;
	}
}

File: app/src/main/java/de/markusfisch/android/shadereditor/hardware/PressureListener.kt
package de.markusfisch.android.shadereditor.hardware

import android.content.Context
import android.hardware.Sensor
import android.hardware.SensorEvent

class PressureListener(context: Context) : AbstractListener(context) {
    var pressure: Float = 0f
        private set

    fun register(): Boolean = register(Sensor.TYPE_PRESSURE)

    override fun onSensorChanged(event: SensorEvent) {
        pressure = event.values[0]
    }
}



